op-e2e/e2eutils/geth/wait.go

```diff
@@ -83,25 +86,80 @@ func WaitForTransaction(hash common.Hash, client *ethclient.Client, timeout time
 	}
 }
 
-func WaitForBlock(number *big.Int, client *ethclient.Client, timeout time.Duration) (*types.Block, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
+type waitForBlockOptions struct {
+	noChangeTimeout time.Duration
+	absoluteTimeout time.Duration
+}
 
-	headChan := make(chan *types.Header, 100)
-	headSub, err := client.SubscribeNewHead(ctx, headChan)
-	if err != nil {
-		return nil, err
+func WithNoChangeTimeout(timeout time.Duration) WaitForBlockOption {
+	return func(o *waitForBlockOptions) {
+		o.noChangeTimeout = timeout
 	}
-	defer headSub.Unsubscribe()
+}
+
+func WithAbsoluteTimeout(timeout time.Duration) WaitForBlockOption {
+	return func(o *waitForBlockOptions) {
+		o.absoluteTimeout = timeout
+	}
+}
+
+type WaitForBlockOption func(*waitForBlockOptions)
+
+// WaitForBlock waits for the chain to advance to the provided block number. It can be configured with
+// two different timeout: an absolute timeout, and a no change timeout. The absolute timeout caps
+// the maximum amount of time this method will run. The no change timeout will return an error if the
+// block number does not change within that time window. This is useful to bail out early in the event
+// of a stuck chain, but allow things to continue if the chain is still advancing.
+//
```

Comment:
If used improperly, I think it's possible to increase flake failures. For tests which might see sporadic updates, a `noChangeTimeout` could trip when a big/slow block comes through. I'm not sure if we have any tests like that today, but in any case it will be the responsibility of the caller to correctly configure their Wait Options. NOOP for this PR, just observing.

Reply:
Are there blocks in tests that take more than 30 second to import?

Reply:
No, I don't think there's anything that really expects this behavior now. Just thinking about future callers who may not exist :P